Builds fail on Minix since v5.31.6

<!--
Note: you can also replace the whole content with a file generated by the perlbug utility;
perlbug reports by email are no longer supported.
Be sure to enclose your perl configuration in a fenced code block to preserve formatting:
https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks
-->
**Description**

I came to build v5.32.0 on Minix (Minix monty 3.4.0 Minix 3.4.0 (GENERIC) i386) and after Configure it failed to build opmini with the following error:
```
monty# make
echo @`sh  cflags "optimize='-O'" opmini.o`  -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLOB opmini.c
@cc -c -DPERL_CORE -fno-strict-aliasing -pipe -fstack-protector-strong -D_FORTIFY_SOURCE=2 -O -Wall -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLOB opmini.c
In file included from op.c:163:
./perl.h:2809:13: error: expected ';' at end of declaration list
        int si_signo;
            ^
/usr/include/sys/siginfo.h:139:23: note: expanded from macro 'si_signo'
#define si_signo        _info._signo
                             ^
1 error generated.
*** Error code 1

Stop.
make: stopped in /usr/opt/build/perl5
```

v5.30.3 built fine

I have bisected the build failure to the commit e712489

**Steps to Reproduce**

I have been using the following Configure arguments:
```
./Configure -des -Uinstallusrbinperl -Uuseithreads -Uusemymalloc -Dldflags='-Wl,-rpath,/usr/pkg/lib -fstack-protector-strong -L/usr/pkg/lib' -Dlibpth='/usr/pkg/lib /usr/include/clang-3.6 /usr/lib /lib' -Dusedevel
```
**Expected behavior**

I expected perl to build successfully.

**Perl configuration**
This is the output from the v5.30.3 that I successfully built and installed, as I couldn't get v5.32.0 to build.
```
monty# /opt/perl-5.30.3/bin/perl -V
Summary of my perl5 (revision 5 version 30 subversion 3) configuration:
   
  Platform:
    osname=minix
    osvers=3.4.0
    archname=i386-minix
    uname='minix monty 3.4.0 minix 3.4.0 (generic) i386 '
    config_args='-e -Uuseithreads -Uusemymalloc -Dprefix=/opt/perl-5.30.3'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=undef
    use64bitall=undef
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-fno-strict-aliasing -pipe -fstack-protector-strong -D_FORTIFY_SOURCE=2'
    optimize='-O'
    cppflags='-fno-strict-aliasing -pipe -fstack-protector-strong'
    ccversion=''
    gccversion='4.2.1 Compatible Clang 3.6 (branches/release_36 237755)'
    gccosandvers=''
    intsize=4
    longsize=4
    ptrsize=4
    doublesize=8
    byteorder=1234
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=12
    longdblkind=3
    ivtype='long'
    ivsize=4
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=4
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags ='-Wl,-rpath,/usr/pkg/lib -fstack-protector-strong -L/usr/pkg/lib'
    libpth=/usr/pkg/lib /usr/include/clang-3.6 /usr/lib /lib
    libs=-lgdbm -lm -lcrypt -lutil -lc
    perllibs=-lm -lcrypt -lutil -lc
    libc=/usr/lib/libc.so
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E '
    cccdlflags='-DPIC -fPIC '
    lddlflags='-shared  -L/usr/pkg/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
  Built under minix
  Compiled at Jun 28 2020 11:42:34
  @INC:
    /opt/perl-5.30.3/lib/site_perl/5.30.3/i386-minix
    /opt/perl-5.30.3/lib/site_perl/5.30.3
    /opt/perl-5.30.3/lib/5.30.3/i386-minix
    /opt/perl-5.30.3/lib/5.30.3
```

--- a/perl.h
+++ b/perl.h
@@ -2968,6 +2968,9 @@ typedef struct padname PADNAME;
 #if defined(HAS_SIGACTION) && defined(SA_SIGINFO)
     typedef siginfo_t Siginfo_t;
 #else
+#ifdef si_signo /* minix */
+#undef si_signo
+#endif
     typedef struct {
         int si_signo;
     } Siginfo_t;

